flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroDeserializationSchema.java

```diff
@@ -34,14 +35,24 @@ public class AvroDeserializationSchema
         implements BigQueryDeserializationSchema<GenericRecord, GenericRecord> {
 
     private final String avroSchemaString;
+    private static final Logger LOG = LoggerFactory.getLogger(AvroDeserializationSchema.class);
 
     public AvroDeserializationSchema(String avroSchemaString) {
         this.avroSchemaString = avroSchemaString;
     }
 
     @Override
-    public GenericRecord deserialize(GenericRecord record) throws IOException {
-        return record;
+    public GenericRecord deserialize(GenericRecord record) throws BigQueryConnectorException {
+        try {
+            return record;
+        } catch (RuntimeException e) {
+            LOG.error(
+                    String.format(
+                            "Error deserializing Avro Generic Record %s to Avro Generic Record.%nError: %s.%nCause:%s ",
+                            record.toString(), e.getMessage(), e.getCause()));
+            throw new BigQueryConnectorException(
+                    "Error in deserializing to Avro Generic Record", e);
+        }
     }
```

Comment:
Why does this method even exist? It returns its argument. How is any error possible?

Reply:
I understand we need to override the interface, but why would you wrap `return record;` in a try block and catch exceptions?

Reply:
Removing the wrap. Error is possible only in `AvroToRowDataDeserializationSchema`